test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/ManagedKieServersWithSmartRouterAndControllerIntegrationTest.java

```diff
@@ -0,0 +1,293 @@
+
+        List<TaskSummary> activeTasks = smartTaskServicesClient.findTasks(USER_YODA, 0, 10);
+        assertThat(activeTasks).hasSize(2);
+        assertThat(activeTasks.get(0).getContainerId()).isEqualTo(CONTAINER_ID_GHI);
+        assertThat(activeTasks.get(0).getProcessInstanceId()).isEqualTo(signalProcessInstances.get(4));
+        assertThat(activeTasks.get(1).getContainerId()).isEqualTo(CONTAINER_ID_GHI);
+        assertThat(activeTasks.get(1).getProcessInstanceId()).isEqualTo(signalProcessInstances.get(5));
+
+        smartTaskServicesClient.startTask(CONTAINER_ID_GHI, activeTasks.get(0).getId(), USER_YODA);
+        kieServerGHITaskServicesClient.startTask(CONTAINER_ID_GHI, activeTasks.get(1).getId(), USER_YODA);
+        kieServerGHITaskServicesClient.completeTask(CONTAINER_ID_GHI, activeTasks.get(0).getId(), USER_YODA, Collections.EMPTY_MAP);
+        smartTaskServicesClient.completeTask(CONTAINER_ID_GHI, activeTasks.get(1).getId(), USER_YODA, Collections.EMPTY_MAP);
+
+        assertThat(smartQueryServicesClient.findProcessesByContainerId(CONTAINER_ID_GHI, 0, 10)).hasSize(2);
+        smartProcessServicesClient.abortProcessInstance(CONTAINER_ID_DEF, signalProcessInstances.get(3));
+
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED, org.kie.api.runtime.process.ProcessInstance.STATE_ABORTED), 0, 10)).hasSize(6);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE), 0, 10)).hasSize(0);
+        assertThat(smartTaskServicesClient.findTasks(USER_YODA, 0, 10)).hasSize(0);
+    }
+
+    private List<Long> createSignalProcesses() {
+        List<Long> signalProcessInstances = new ArrayList<>();
+
+        signalProcessInstances.add(smartProcessServicesClient.startProcess(CONTAINER_ID_ABC, SIGNALUSERTASK_PROCESS_ID));
+        signalProcessInstances.add(kieServerABCProcessServicesClient.startProcess(CONTAINER_ID_ABC, SIGNALUSERTASK_PROCESS_ID));
+
+        signalProcessInstances.add(smartProcessServicesClient.startProcess(CONTAINER_ID_DEF, SIGNALUSERTASK_PROCESS_ID));
+        signalProcessInstances.add(kieServerDEFProcessServicesClient.startProcess(CONTAINER_ID_DEF, SIGNALUSERTASK_PROCESS_ID));
+
+        signalProcessInstances.add(smartProcessServicesClient.startProcess(CONTAINER_ID_GHI, SIGNALUSERTASK_PROCESS_ID));
+        signalProcessInstances.add(kieServerGHIProcessServicesClient.startProcess(CONTAINER_ID_GHI, SIGNALUSERTASK_PROCESS_ID));
+
+        return signalProcessInstances;
+    }
+
+    protected int findServerDeploymentWithName(String kieServerName) {
+        for (int i = 0; i < deploymentScenario.getKieServerDeployments().size(); i++) {
+            if (deploymentScenario.getKieServerDeployments().get(i).getUrl().toString().contains(kieServerName)) {
+                return i;
+            }
+        }
+        return -1;
```

Comment:
This seems to be invalid option which can cause IndexOutOfBoundsException. I think it would be better to fail fast - throwing some runtime exception with clear message.